### PR TITLE
feat(submission) Lambda service from collecting JSON from REST payloa…

### LINKED ISSUE #407: Implement Microservice for collecting App Metrics
## Description

We need a service that will collect metrics from the Covid Alert App this service will need to aggregate the Metrics for a specific period (TBD, but lets start by bucketing per day). 

These aggregate metrics will be stored in a new table in the existing Aurora MySQL Database.

## AC

- An endpoint is created that allows for the submission of metrics from the App
- Calls to this endpoint are only allowed by the app
- The metrics are aggregated per day and are stored in a table in the existing Aurora MySQL Database
- The new services are covered by automated tests
- API Documentation has been updated
- Infrastructure as Code is written for new Service
- CI is updated for new service
  - Linting of new source code (see #418)
  - Snyk scanning of new dependencies
  - New Tests are running on PR
- CNAME record pointing to metric lambda (https://github.com/cds-snc/covid-alert-server-staging-terraform/issues/81)

### PR DESCRIPTION
**Pull request by tpjods**
> simple micro-service to collect data in JSON format and save it as a unique object in to S3. This is a MVP solution that will accept any JSON file. The service will create a UUID for each new object.  close #407 Implement Microservices for collecting metrics.

This is a basic data collector. 

Requires 2 environment variables. 
"dataBucket"   <-- The name of the bucket you are saving to
"fileLoca" <-- The folder in the bucket you are saving to

It also requires an inline policy to be added to the lambda function providing putObject permissions on the bucket. If this is an encrypted bucket the KMS encrypt policy is also required.

### CODE REVIEW

**Review by CalvinRodo — CHANGES_REQUESTED** [review]
An automated test is also required for this piece of code, we use Jest as our JS Unit Testing Framework.

I took a look at your other PR you submitted I'm not seeing the issues there that are present here: https://github.com/cds-snc/covid-alert-server-qa-tools/pull/1

File: serverless/create-record/index.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    uuid = require( 'uuid' ),
+    S3  = new AWS.S3(),
+    fs = require( 'fs' );
+
+exports.handler = async (event, context) => {
+
+    let errorMessage = "Malformed Package";
+    let transactionStatus = "FAILED";
+    let bucket = process.env.dataBucket;           //encrypted env_var
+    let fP = process.env.fileLoca;                  //encrypted env_var
```
Comment (wmoussa-cds) [inline serverless/create-record/index.js:27]: fileLoca looks like a typo?

File: serverless/create-record/index.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    let d = new Date();
```
Comment (wmoussa-cds) [inline serverless/create-record/index.js:28]: d as a variable name?

File: serverless/create-record/index.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    let filename = uuid.v1();  // RFC4122 standard uuid Timestamp based generation
+
+
+// informational for DEV
```
Comment (wmoussa-cds) [inline serverless/create-record/index.js:32]: inline comments as such are not useful, lets keep the code as clean as possible (and readable)

File: serverless/create-record/index.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+console.log("event", event);
+console.log("event json : ", JSON.stringify(event));
+
+	var body = JSON.stringify(event);
```
Comment (wmoussa-cds) [inline serverless/create-record/index.js:36]: indent properly

File: serverless/create-record/index.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    fs = require( 'fs' );
+
+exports.handler = async (event, context) => {
+
+    let errorMessage = "Malformed Package";
+    let transactionStatus = "FAILED";
+    let bucket = process.env.dataBucket;           //encrypted env_var
+    let fP = process.env.fileLoca;                  //encrypted env_var
+    let d = new Date();
+    let filename = uuid.v1();  // RFC4122 standard uuid Timestamp based generation
```
[the remark this answers is not in the document]
Comment (wmoussa-cds) [reply serverless/create-record/index.js:29]: i think most initialized variables in the block above should be const

File: serverless/create-record/index.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+
+// informational for DEV
+console.log("event", event);
+console.log("event json : ", JSON.stringify(event));
+
+	var body = JSON.stringify(event);
+
+    // Is this an attack?
+    if(body.includes("/script" || "< script >" || "alert(" || "SELECT " || "DROP " )){
```
Comment (wmoussa-cds) [inline serverless/create-record/index.js:39]: isn't there a util library we could use instead?

File: serverless/create-record/index.js [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    let filename = uuid.v1();  // RFC4122 standard uuid Timestamp based generation
+
+
+// informational for DEV
+console.log("event", event);
+console.log("event json : ", JSON.stringify(event));
+
+	var body = JSON.stringify(event);
+
+    // Is this an attack?
```
Comment (wmoussa-cds) [inline serverless/create-record/index.js:38]: instead of an inline comment, use a method

File: serverless/create-record/index.js [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    if(body.includes("/script" || "< script >" || "alert(" || "SELECT " || "DROP " )){
+        return Error(errorMessage);
+    }
+
+    //Screen out direct service attacks. Sanitize inputs
+    else{
```
Comment (wmoussa-cds) [inline serverless/create-record/index.js:44]: bad formatting

**Comment by CalvinRodo** [comment]
Closing in favour of https://github.com/cds-snc/covid-alert-serverless/pull/2

**Pull request closed without merging**